libnavigator/src/main/java/com/mapbox/navigation/navigator/MapboxNativeNavigator.kt

```diff
@@ -23,49 +26,217 @@ interface MapboxNativeNavigator {
+
+    /**
+     * Uses valhalla and local tile data to generate mapbox-directions-api-like json
+     *
+     * @param url the directions-based uri used when hitting the http service
+     * @return a [RouterResult] object with the json and a success/fail bool
+     */
     fun getRoute(url: String): RouterResult
+
+    /**
+     * Passes in an input path to the tar file and output path.
+     *
+     * @param tarPath The path to the packed tiles.
+     * @param destinationPath The path to the unpacked files.
+     *
+     * @return the number of unpacked tiles
+     */
     fun unpackTiles(tarPath: String, destinationPath: String): Long
+
+    /**
+     * Removes tiles wholly within the supplied bounding box. If the tile is not
+     * contained completely within the bounding box it will remain in the cache.
+     * After removing files from the cache any routers should be reconfigured
+     * to synchronize their in memory cache with the disk
+     *
+     * @param tilePath The path to the tiles.
+     * @param southwest The lower left coord of the bbox.
+     * @param northeast The upper right coord of the bbox.
+     *
+     * @return the number of tiles removed
+     */
     fun removeTiles(tilePath: String, southwest: Point, northeast: Point): Long
 
     // History traces
 
+    /**
+     * Gets the history of state changing calls to the navigator this can be used to
+     * replay a sequence of events for the purpose of bug fixing
+     *
+     * @return a json representing the series of events that happened since the last time
+     * history was toggled on
+     */
     fun getHistory(): String
+
+    /**
+     * Toggles the recording of history on or off.
+     * Toggling will reset all history call [getHistory] first before toggling to retain a copy
+     *
+     * @param isEnabled set this to true to turn on history recording and false to turn it off
+     */
     fun toggleHistory(isEnabled: Boolean)
+
+    /**
+     * Adds a custom event to the navigators history. This can be useful to log things that
+     * happen during navigation that are specific to your application.
+     * @param eventType the event type in the events log for your custom even
+     * @param eventJson the json to attach to the "properties" key of the event
+     */
     fun addHistoryEvent(eventType: String, eventJsonProperties: String)
 
     // Configuration
 
+    /**
+     * Gets the current configuration used for navigation.
+     *
+     * @return the [NavigatorConfig] used for navigation.
+     */
     fun getConfig(): NavigatorConfig
+
+    /**
+     * Updates the configuration used for navigation. Passing null resets the config
+     *
+     * @param config the new [NavigatorConfig]
+     */
     fun setConfig(config: NavigatorConfig?)
 
     // Other
 
+    /**
+     * Gets the voice instruction at a specific step index in the route. If there is no
+     * voice instruction at the specified index, we will do something.
```

Comment:
```suggestion
     * voice instruction at the specified index, we won't return anything.
```

Possibly? :)

Reply:
Hah) Right, I copied it from native docs, and didn't check this one) Thanks for your alertness!